src/coreclr/debug/createdump/dumpwritermacho.cpp

```diff
@@ -0,0 +1,281 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "createdump.h"
+#include "specialthreadinfo.h"
+
+// Write the core dump file
+bool
+DumpWriter::WriteDump()
+{
+    BuildSegmentLoadCommands();
+
+    BuildThreadLoadCommands();
+
+    uint64_t fileOffset = 0;
+    if (!WriteHeader(&fileOffset)) {
+        return false;
+    }
+
+    TRACE("Writing %zd thread commands to core file\n", m_threadLoadCommands.size());
+
+    // Write thread commands
+    for (const ThreadCommand& thread : m_threadLoadCommands)
```

Comment:
I write out the thread commands first only because the lldb save-core (lldb has code to write Macho core dumps but not Linux) code does it that way. I don't think it matters either way.